solvers/relay_constellation/mclp_teg_contact_plan/src/mclp.py

```
@@ -0,0 +1,440 @@
+    link_records: Iterable[LinkRecord],
+    max_candidates_for_milp: int = 20,
+    max_added_for_milp: int = 5,
+    time_limit_seconds: float = 30.0,
+) -> tuple[list[CandidateSatellite], dict[str, object]] | None:
+    """Optional small MILP MCLP using PuLP/CBC.
+
+    Returns None if problem is too large or PuLP is unavailable.
+    """
+    try:
+        import pulp
+    except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)